vocabulary-draft/continuous-delivery.md

```diff
@@ -0,0 +1,114 @@
+
+Due the dynamic nature of Pipelines, most of actual work needs to be queued to happen in a distributed way, hence Queued events are added. Adopters can choose to ignore these events if they don't apply to their use cases. 
+
+- **PipelineRun Queued**: a PipelineRun has been schedule to run
+- **PipelineRun Started**: a PipelineRun has started and it is running
+- **PipelineRun Finished**: a PipelineRun has finished it execution, the event will contain the finished status, success, error or failure
+
+Each pipeline is defined as a set of Tasks to be performed in sequence, hence tracking this sequence might be important for some cases. A TaskRun is an instance of the Task defined inside the pipeline, as you can expect multiple execution of the pipelines (each a PipelineRun) you can also expect multiple execution of the Tasks, for that reason we use TaskRun to refer to one of these instances.  
+
+- **TaskRun Started**: a TaskRun inside a PipelineRun has started.
+- **TaskRun Finished**: a TaskRun inside a PipelineRun has finished. 
+
+Pipeline Events MUST include the following attributes:
+- **Event Type**: the type is restricted to include `CD.Pipeline**` prefix. For example `CD.Pipeline.PipelineRun.Queued` or `CD.Pipeline.Tests.Started`
+- **PipelineRun Id**: unique identifier for a pipeline execution
+- **Pipeline Name**: unique identifier for the pipeline, not for the instance. A pipeline can have multiple instances/runs.  
+- **PipelineRun Status**: current status of the PipelineRun at the time when the event was emitted. If the pipeline is finished, this attribute should reflect if it finished successfully or if there was an error on the execution. Possible statuses: [Running, Finished, Error]
+
+Optional attributes: 
+- **PipelineRun URL**: URL to locate where  pipeline instances are running
+- **PipelineRun Errors**: error field to indicate possible compilation, test, build and package errors.
+
+## Product Composition Events
+
+When the products delivered to users can be, or needs to be, broken down into components (hardware, software, configuration files etc.), the creation of versions of said components and the compositoning of components into products or product subsystems may warrant the use of separate events detailed in this section.
+
+Product Composition Events MUST include the following attributes:
+
+- **Event Type**: the type is restricted to include `CD.ProductComposition` prefix. For example `CD.ProductComposition.ComponentVersionCreated` or `CD.ProductComposition.CompositionCreated`
+
+### Component Version Created
+
+Announces that a new version of a Product Component has been created.
+
+Examples of Component Versions include:
+
+- New version of a software module or application.
+- New version of a document or piece of configuration.
+- New revision of a hardware component.
+
+The purpose of this event would be for systems to start creating Product Compositions / Baseline Versions that include this Component Version.
```

Comment:
@erkist while I think that we need to deal with composition, it feel to me that we are still not quite clear on the basic building blocks, so I would prefer to focus on those basic building blocks first and then add composition. For example, if we want to group artifacts/components together, we can always include metadata to reflect the parent/child relationship.

Reply:
@salaboy I see your point. As mentioned above though, it becomes tricky once hardware comes into the mix. But I am fine with waiting with those parts. as long as they make it into the spec at some point.